Source/Scene/PickDepth.js

```diff
@@ -0,0 +1,169 @@
+        if (supported) {
+            updateCopyCommands(this, context);
+        }
+
+        this._supported = supported;
+    };
+
+    defineProperties(PickDepth.prototype, {
+        supported : {
+            get : function() {
+                return this._supported;
+            }
+        }
+    });
+
+    function destroyTextures(pickDepth) {
+        pickDepth._depthTexture = pickDepth._depthTexture && !pickDepth._depthTexture.isDestroyed() && pickDepth._depthTexture.destroy();
+    }
+
+    function destroyFramebuffers(pickDepth) {
+        pickDepth.framebuffer = pickDepth.framebuffer && !pickDepth.framebuffer.isDestroyed() && pickDepth.framebuffer.destroy();
+    }
+
+    function createTextures(pickDepth, context, width, height) {
+        pickDepth._depthTexture = context.createTexture2D({
+            width : width,
+            height : height,
+            pixelFormat : PixelFormat.RGBA,
+            pixelDatatype : PixelDatatype.UNSIGNED_BYTE
+        });
+    }
+
+    function createFramebuffers(pickDepth, context, width, height) {
+        destroyTextures(pickDepth);
+        destroyFramebuffers(pickDepth);
+
+        createTextures(pickDepth, context, width, height);
+
+        pickDepth.framebuffer = context.createFramebuffer({
+            colorTextures : [pickDepth._depthTexture],
+            destroyAttachments : false
+        });
+
+        if (pickDepth.framebuffer.status !== WebGLRenderingContext.FRAMEBUFFER_COMPLETE) {
+            destroyTextures(pickDepth);
+            destroyFramebuffers(pickDepth);
+            return false;
+        }
+
+        return true;
+    }
+
+    function updateFramebuffers(pickDepth, context, depthTexture) {
+        if (!context.depthTexture) {
+            return;
+        }
+
+        var width;
+        var height;
+
+        if (defined(depthTexture)) {
+            width = depthTexture.width;
+            height = depthTexture.height;
+        } else {
+            width = context.drawingBufferWidth;
+            height = context.drawingBufferHeight;
+        }
+
+        var texture = pickDepth._depthTexture;
+        var textureChanged = !defined(texture) || texture.width !== width || texture.height !== height;
+        if (!defined(pickDepth.framebuffer) || textureChanged) {
+            if (!createFramebuffers(pickDepth, context, width, height)) {
+                // framebuffer creation failed
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    function updateCopyCommands(pickDepth, context, depthTexture) {
+        if (!defined(pickDepth._copyDepthCommand)) {
+            var fs = 'uniform sampler2D u_texture;\n' +
+                     'varying vec2 v_textureCoordinates;\n' +
+                     'void main()\n' +
+                     '{\n' +
+                     '    float depth = texture2D(u_texture, v_textureCoordinates).r;\n' +
+                     '    float temp = depth * 255.0;\n' +
```

Comment:
Think this is generic enough for a `czm_` function, e.g., for a shadow mapping fallback?